File: julia/day01.jl
#!/usr/bin/env julia

function part1(m)::Int
    m ÷ 3 - 2
end

function part2(m)::Int
    s = 0

    m = part1(m)
    while m > 0
        s += m
        m = part1(m)
    end

    return s
end

d = [parse(Int, x) for x in readlines()]

total = sum(part1, d)
total_fuel = sum(part2, d)


using Printf
@printf("part1: %d\n", total)
@printf("part2: %d\n", total_fuel)

File: python/day01.py
#!/usr/bin/env python3

import fileinput

def part1(m):
    return m // 3 - 2

def part2(m):
    s = 0

    m = part1(m)
    while m > 0:
        s += m
        m = part1(m)

    return s

def main():

    d = [int(line) for line in fileinput.input()]
    total = sum(map(part1, d))
    total_fuel = sum(map(part2, d))

    print(f'part1: {total}')
    print(f'part2: {total_fuel}')

if __name__ == '__main__':
    main()
